AY2122S1-CS2103T-T11-2/tp
# Walking skeleton

### Description



This PR includes a major rework of the original AddressBook3 codebase to form a fully functional walking skeleton for the `LingoGO!` application. All existing components and tests have been successful migrated, including the JavaFX GUI.

Documentation and individual project portfolio will be updated in separate pull requests.

### Type of change


### How to test


- [ ] Read through the unit and integration tests to spot any gaps in testing
- [ ] Manually build the jar file and use it to test the following:
- [ ] Manually test the existing functionality in the new LingoGO! application
    - `add`
    - `clear`
    - `delete`
    - `edit`
    - `exit`
    - `find`
    - `help`
    - `list`
- [ ] Test the `Storage` functionality in the app
    - Try saving something and reloading it
    - Deleting the save file and re-running the app should generate a default set of 4 flashcards in the 'new' save file
- [ ] Try to break the parser
    - random Unicode characters from other languages
    - illegal commands and arguments
    - extra whitespaces or lack thereof

### Checklist



- [ ] I have updated the Developer Guide
- [ ] I have updated my individual project portfolio
